kratos/containers/csr_matrix.h

```diff
@@ -696,9 +689,9 @@ class CsrMatrix
     ///@name Member Variables
     ///@{
     bool mIsOwnerOfData = true;
-    IndexType* mpRowIndicesData = nullptr;
-    IndexType* mpColIndicesData = nullptr;
-    TDataType* mpValuesVectorData = nullptr;
+    Kratos::unique_ptr<IndexType[]> mpRowIndicesData; //this is the pointer who "owns" the data. DO NOT USE IT as such (it may be uninitialized if data is not owned), only use the span wrapper just below
```

Comment:
the other "problematic" thing is that while the span, for example mRowIndices, is guaranteed to point to the correct data, mpRowIndicesData may be null (so not refer to the same data) in the case in which the data is not owned